src/main/java/com/direwolf20/buildinggadgets/common/building/modes/GridMode.java

```diff
@@ -0,0 +1,42 @@
+package com.direwolf20.buildinggadgets.common.building.modes;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.building.IPlacementSequence;
+import com.direwolf20.buildinggadgets.common.building.IValidatorFactory;
+import com.direwolf20.buildinggadgets.common.building.placement.Grid;
+import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
+import com.direwolf20.buildinggadgets.common.tools.IAtopSupport;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+
+public class GridMode extends AbstractMode implements IAtopSupport {
+
+    //TODO give config option
```

Comment:
When?

Reply:
next PR in 1.12 -- don't want add any extra featurea into this one.